core/domain/suggestion_registry.py

```diff
@@ -635,6 +635,209 @@ def pre_update_validate(self, change):
+                    self.content))
+
+    def accept(self):
+        """Accepts the voiceover application. Each subclass must implement this
+        function.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseVoiceoverApplication should implement accept.')
+
+    def reject(self):
+        """Rejects the voiceover application. Each subclass must implement this
+        function.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseVoiceoverApplication should implement reject.')
+
+    @property
+    def is_handled(self):
+        """Returns true if the voiceover application has either been accepted or
+        rejected.
+
+        Returns:
+            bool. Whether the voiceover application has been handled or not.
+        """
+        return self.status != suggestion_models.STATUS_IN_REVIEW
+
+
+class ExplorationVoiceoverApplication(BaseVoiceoverApplication):
+    """Domain object for a voiceover application for exploration."""
+
+    def __init__( # pylint: disable=super-init-not-called
+            self, voiceover_application_id, target_id, status, author_id,
+            final_reviewer_id, language_code, filename, content,
+            rejection_message):
+        """Initializes a ExplorationVoiceoverApplication domain object.
+
+        Args:
+            voiceover_application_id: str. The ID of the voiceover application.
+            target_id: str. The ID of the target entity.
+            status: str. The status of the voiceover application.
+            author_id: str. The ID of the user who submitted the voiceover
+                application.
+            final_reviewer_id: str|None. The ID of the reviewer who has
+                accepted/rejected the voiceover application.
+            language_code: str. The language code for the voiceover application.
+            filename: str. The filename of the voiceover audio.
+            content: str. The html content which is voiceover in the
+                application.
+            rejection_message: str. The plain text message submitted by the
+                reviewer while rejecting the application.
+        """
+        self.voiceover_application_id = voiceover_application_id
+        self.target_type = suggestion_models.TARGET_TYPE_EXPLORATION
+        self.target_id = target_id
+        self.status = status
+        self.author_id = author_id
+        self.final_reviewer_id = final_reviewer_id
+        self.language_code = language_code
+        self.filename = filename
+        self.content = content
+        self.rejection_message = rejection_message
+
+    def accept(self, reviewer_id):
+        """Accepts the voiceover application and updates the final_reviewer_id.
+
+        Args:
+            reviewer_id: str. The user ID of the reviewer.
+        """
+        self.final_reviewer_id = reviewer_id
+        self.status = suggestion_models.STATUS_ACCEPTED
+        self.validate()
```

Comment:
Validate before changing the status as otherwise you can end up with an accepted invalid suggestion.

Reply:
Also, shouldn't the model be updated here?

Reply:
> Validate before changing the status as otherwise, you can end up with an accepted invalid suggestion.

We are not changing the model here so I don't think we need to check this before updating the domain object! 

In general, I think we should validate a domain object after changing the properties of the object!

Reply:
> Also, shouldn't the model be updated here?

This is the domain layer so I don't think we need to touch services/models in this layer!

Reply:
Oh I'm sorry... What I meant was shouldn't there be a corresponding service function which actually puts that change into the database... Just changing the domain object doesn't really help right?

Reply:
> We are not changing the model here so I don't think we need to check this before updating the domain object!

Assuming that every other change made to the object is validated, this is fine. But I think it would be better to do it anyway to be safe.

Reply:
> But I think it would be better to do it anyway to be safe.

I'm not sure why I should validate the object before doing any changes here, In general, none of the update methods use to have validation like if you check the exp_domain.py update_* methods they only do what the method name explains and the validation is called explicitly in the service layer after calling the update_* method.

Reply:
Hmmm, ok. I would have liked to see the service layer along with this function. But if you plan to add that later, then fine!